framework/include/media/FocusManager.h

```diff
@@ -0,0 +1,102 @@
+/* ****************************************************************
+ *
+ * Copyright 2018 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ******************************************************************/
+
+/**
+ * @ingroup MEDIA
+ * @{
+ */
+
+/**
+ * @file media/FocusManager.h
+ * @brief Media FocusManager APIs
+ */
+
+#ifndef __MEDIA_FOCUSMANAGER_H
+#define __MEDIA_FOCUSMANAGER_H
+
+#include <unistd.h>
+#include <list>
+#include <memory>
+#include <mutex>
+
+namespace media {
+static const int FOCUS_NONE = 0;
+static const int FOCUS_GAIN = 1;
+static const int FOCUS_LOSS = -1;
+
+class FocusChangeListener
+{
+public:
+	virtual void onFocusChange(int focusChange) {};
+};
+
+class FocusRequest
+{
+public:
+	FocusRequest(int id, std::shared_ptr<FocusChangeListener> listener);
+	int getId();
+	std::shared_ptr<FocusChangeListener> getListener();
+
+private:
+	int mId;
+	std::shared_ptr<FocusChangeListener> mListener;
+};
+
+class FocusRequestBuilder
```

Comment:
In future, there's chance that apply builder for datasource.
Then we need make generic builder which can be inherited.
Please consider it.

Reply:
I mean do not implement it now, in future, I mean